src/test/java/StringManipulationTest.java

```diff
@@ -26,19 +25,29 @@ public void testCount1() {
         assertEquals(4, length);
     }
 
+    // check to see that a null string will return a word count of 0.
     @Test
     public void testCount2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString(null);
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    // tests if numbers or special characters will mess with the word count
     @Test
     public void testCount3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a string that the number $40.00 in the string");
+        int length = manipulatedstring.count();
+        assertEquals(11, length);
     }
 
+    // another test, but this time we add quotation marks to see if it messes with
+    // word count
     @Test
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This a string that can't fail just because \"can't\" is in the string");
```

Comment:
This test seems redundant with Test case #1. Can it be replaced with testing extra whitespace (EX: " abc") and the one where string has more than 1 whitespace between words (EX: "abc  def")?